www/docs/migration/migrate-from-v9-to-v10.mdx

```diff
@@ -171,21 +138,58 @@ const appRouter = t.router({
 });
 ```
 
-## Migrating from v9
+</details>
 
-:::info
-There are [a few features that are not supported by `.interop()`](#limitations-of-interop). We expect nearly all of our users to be able to use `.interop()` to migrate their server side code in only a few minutes. If you are discovering that `.interop()` is not working correctly for you, be sure to [check here](#limitations-of-interop).
-:::
+Here's a full example of how the `t` variable may look like in v10, including a data transformer, OpenAPI metadata, and an error formatter:
 
-You may choose to start writing new procedures using the v10 syntax - but keep your v9 procedures in place for now. Incremental adoption is only a matter of merging the two routers together.
+```ts title='/src/server/trpc.ts'
+import { initTRPC } from '@trpc/server';
+import superjson from 'superjson';
 
-Turning your v9 router into a v10 router only takes 10 characters. Add `.interop()` to the end of your v9 router...and you're done with your server code!
+// Context is usually inferred,
+// but we will need it here for this example.
+interface Context {
+  user?: {
+    id: string;
+    name: string;
+  };
+}
 
-We will create a simple hello world v10 router to demonstrate interoperability mode. If you have not learned how to create a more complex tRPC v10 router yet, we encourage you to check out the rest of the documentation.
+interface Meta {
+  openapi: {
+    enabled: boolean;
+    method: string;
+    path: string;
+  };
+}
 
-### 1. Enable `interop()` on a v9 router
+export const t = initTRPC
+  .context<Context>()
+  .meta<Meta>()
+  .create({
+    errorFormatter({ shape, error }) {
+      return {
+        ...shape,
+        data: {
+          ...shape.data,
+          zodError:
+            error.code === 'BAD_REQUEST' && error.cause instanceof ZodError
+              ? error.cause.flatten()
+              : null,
+        },
+      };
+    },
+    transformer: superjson,
+  });
+```
+
```

Comment:
Do we want all of this ina disclosure too maybe?

Reply:
I'm in for it!